### PR TITLE
add `mountPoints` to `Service`

### PR DESCRIPTION
**Pull request by zvictor**
Access to Containers' `mountPoints` is needed in order to attach volumes/filesystems.

This is part of sst/sst#4616 and [this](https://discord.com/channels/983865673656705025/1285733025547817052).

* https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specify-efs-config.html
* https://docs.aws.amazon.com/AmazonECS/latest/developerguide/specify-volume-config.html

### CODE REVIEW

File: platform/src/components/aws/cluster.ts [hunk 1]
```diff
@@ -1138,7 +1163,7 @@ export class Cluster extends Component {
   constructor(
     name: string,
     args: ClusterArgs,
-    opts?: ComponentResourceOptions,
+    opts?: ComponentResourceOptions
```
Comment (zvictor) [inline platform/src/components/aws/cluster.ts]:
```suggestion
    opts?: ComponentResourceOptions,
```

File: platform/src/components/aws/cluster.ts [hunk 2]
```diff
@@ -1168,7 +1193,7 @@ export class Cluster extends Component {
 
     function createCluster() {
       return new ecs.Cluster(
-        ...transform(args.transform?.cluster, `${name}Cluster`, {}, { parent }),
+        ...transform(args.transform?.cluster, `${name}Cluster`, {}, { parent })
```
Comment (zvictor) [inline platform/src/components/aws/cluster.ts]:
```suggestion
        ...transform(args.transform?.cluster, `${name}Cluster`, {}, { parent }),
```

File: platform/src/components/aws/cluster.ts [hunk 3]
```diff
@@ -979,6 +979,31 @@ export interface ClusterServiceArgs {
...
+
+      /**
+       * The path in the container where the volume will be mounted.
+       */
+      containerPath: Input<string>;
+      /**
+       * If this value is true, the container has read-only access to the volume. If this value is false, then the container can write to the volume. The default value is false.
+       */
+      readOnly?: Input<boolean>;
+    }[];
```
Comment (zvictor) [inline platform/src/components/aws/cluster.ts:1006]: should I copy this block to top-level `ServiceArgs`?

**Comment by jayair** [comment]
I see your other PR as well. Maybe we should reopen your old issue and link these there.

**Comment by bjornlll** [comment]
Amazing! Any chance this might get merged soon?

**Comment by fwang** [comment]
Added EFS support in v3.2.36.
- EFS doc https://sst.dev/docs/component/aws/efs
- Cluster doc on configuring EFS https://sst.dev/docs/component/aws/cluster#volumes

**Comment by fwang** [comment]
We are moving sst/ion -> sst/sst and unfortunately this is going to be a messy process because of github's limitations.

i'm going to close this PR for now and it can be re-opened in the new sst/sst repo that will be ready in the next few days if still relevant

**Pull request closed without merging**